### ISSUE TITLE
Bump https-proxy-agent from `5.0.1` to `7.0.1` (or later)

### ISSUE THREAD

**Issue by Nino-K** [rancher-sandbox/rancher-desktop#5057]
In `1.9` we bumped `proxy-agent` from `5.0.1` to `6.1.0` and that caused some unhandled exceptions on the startup as reported by @byjrack, more details [here](https://github.com/rancher-sandbox/rancher-desktop/issues/5009). This turns out to be a [bug](https://github.com/TooTallNate/proxy-agents/issues/207) with the upstream code and requires a PR to address the issue. I'm creating this issue to keep track of the progress on the PR prior to bumping the `proxy-agent` again.

**Comment by gaktive** [comment]
Waiting for upstream bug fix before re-upgrading.

**Comment by byjrack** [comment]
Hopefully not too far off.  https://github.com/TooTallNate/proxy-agents/pull/212

**Comment by jandubois** [comment]
The fix has been released in https://github.com/TooTallNate/proxy-agents/releases/tag/https-proxy-agent%407.0.1 a week ago, and is available from npm at https://www.npmjs.com/package/https-proxy-agent/v/7.0.1.

Why is there no dependabot PR to bump it again?

Since we needed to apply additional changes anyways (see e21c9c2), maybe we should just bump manually. But we still need to understand why we are not getting notified automatically: I do see PRs from the weekly npm check, and no exceptions for this package in `.github/dependabot.yml`.
